model/configuration.py

```diff
@@ -769,3 +774,567 @@ def excluded_audio_data_sources(cls, _db):
+    def key(self):
+        """Returns the setting's key
+
+        :return: Setting's key
+        :rtype: string
+        """
+        return self._key
+
+    @property
+    def label(self):
+        """Returns the setting's label
+
+        :return: Setting's label
+        :rtype: string
+        """
+        return self._label
+
+    @property
+    def description(self):
+        """Returns the setting's description
+
+        :return: Setting's description
+        :rtype: string
+        """
+        return self._description
+
+    @property
+    def type(self):
+        """Returns the setting's type
+
+        :return: Setting's type
+        :rtype: string
+        """
+        return self._type
+
+    @property
+    def required(self):
+        """Returns the boolean value indicating whether the setting is required or not
+
+        :return: Boolean value indicating whether the setting is required or not
+        :rtype: string
+        """
+        return self._required
+
+    @property
+    def default(self):
+        """Returns the setting's default value
+
+        :return: Setting's default value
+        :rtype: string
+        """
+        return self._default
+
+    @property
+    def options(self):
+        """Returns the setting's options (used in the case of select)
+
+        :return: Setting's options (used in the case of select)
+        :rtype: string
+        """
+        return self._options
+
+    @property
+    def category(self):
+        """Returns the setting's category
+
+        :return: Setting's category
+        :rtype: string
+        """
+        return self._category
+
+    @staticmethod
+    def get_configuration_metadata(cls):
+        """Returns a list of 2-tuples containing information ConfigurationMetadata properties in the specified class
+
+        :param cls: Class
+        :type cls: type
+
+        :return: List of 2-tuples containing information ConfigurationMetadata properties in the specified class
+        :rtype: List[Tuple[string, ConfigurationMetadata]]
+        """
+        members = inspect.getmembers(cls)
+
+        for name, member in members:
+            if isinstance(member, ConfigurationMetadata):
+                yield name, member
+
+    def to_settings(self):
+        return {
+            ConfigurationAttribute.KEY.value: self.key,
+            ConfigurationAttribute.LABEL.value: self.label,
+            ConfigurationAttribute.DESCRIPTION.value: self.description,
+            ConfigurationAttribute.TYPE.value: self.type.to_control_type(),
+            ConfigurationAttribute.REQUIRED.value: self.required,
+            ConfigurationAttribute.DEFAULT.value: self.default,
+            ConfigurationAttribute.OPTIONS.value:
+                [option.to_settings() for option in self.options]
+                if self.options
+                else None,
+            ConfigurationAttribute.CATEGORY.value: self.category
+        }
+
+
+class ConfigurationBucket(ConfigurationMetadataOwner):
```

Comment:
Actually since this is the base mixin class, the one we will be seeing throughout  the code, _it_ should be called `HasConfigurationSettings`.

Reply:
Changed it to `ConfigurationGrouping`